modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/snapshot/storage/LastSyncGlobalSnapshotStorage.scala

```diff
@@ -93,9 +94,47 @@ object LastSynchronizedGlobalSnapshotStorage {
           }
         }
 
+      override def getLastNSynchronized(
+        n: Int
+      ): F[Option[List[Hashed[GlobalIncrementalSnapshot]]]] =
+        currencySnapshotStorage.head.flatMap {
+          case Some((_, info)) =>
+            info.globalSnapshotSyncView.flatTraverse { syncView =>
+              val lastOrdinalOpt = syncView.values
+                .map(_.globalSnapshotOrdinal)
+                .groupBy(identity)
+                .toList
+                .sortBy { case (ordinal, occurrences) => (-occurrences.size, ordinal.value.value) }
+                .map { case (ordinal, _) => ordinal }
+                .maxOption
+
+              lastOrdinalOpt match {
+                case Some(lastOrdinal) =>
+                  val selectedOrdinals = (0 until n).flatMap { i =>
+                    NonNegLong.from(lastOrdinal.value.value - i.toLong).toOption.flatMap(SnapshotOrdinal(_))
+                  }.toSet.toList
+
+                  selectedOrdinals.traverse(get).map { results =>
+                    val snapshots = results.flatten
+                    if (snapshots.isEmpty) None else Some(snapshots)
+                  }
+
+                case None =>
+                  Async[F].pure(None)
+              }
```

Comment:
You can just `flatTraverse` and thanks to that you do not need to explicitly return this `None` in lines 122-123:

```scala
              lastOrdinalOpt.flatTraverse { lastOrdinal =>
                val selectedOrdinals = (0 until n).flatMap { i =>
                  NonNegLong.from(lastOrdinal.value.value - i.toLong).toOption.flatMap(SnapshotOrdinal(_))
                }.toSet.toList

                selectedOrdinals
                  .traverse(get)
                  .map { results =>
                    val snapshots = results.flatten
                    if (snapshots.isEmpty) None else Some(snapshots)
                  }
              }
```